packages/wasm-bindings/src/lib.rs

```diff
@@ -7,6 +7,82 @@ use wasm_bindgen::prelude::*;
 #[global_allocator]
 static ALLOC: wee_alloc::WeeAlloc = wee_alloc::WeeAlloc::INIT;
 
+/// Modify the currently selected tool in the document state store
+#[wasm_bindgen]
+pub fn select_tool(tool_id: u32) {
+	todo!()
+}
+
+/// Modify the currently selected Document in the editor state store
+// TODO: should it be possible to open multiple documents at once?
+#[wasm_bindgen]
+pub fn select_document(document_id: u32) {
+	todo!()
+}
+
+/// Load a new .gdd file into the editor
+/// Returns a unique document identifier
+#[wasm_bindgen]
+pub fn load_document(raw_data: &[u8]) -> u32 {
+	todo!()
+}
+
+/// Mouse path drawn on the canvas if the mouse is clicked for a longer period of time
+/// (every mouse position between the mouse down and up event)
+#[wasm_bindgen]
+pub fn canvas_mouse_path(mousepath: &[u64], modkeys: u8, ui_element_id: u32) -> u32 {
```

Comment:
Best way to do this is probably letting JS accumulate mouse pixel movements in a buffer. Then when it gets the chance to call this function with its buffer, the buffer gets purged and it can keep recording afterwards.